instagram-app/src/components/Authentication/withAuthenticate.js

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+const withAuthenticate = Component => Component2 =>
+	class extends React.Component {
+		constructor(props) {
+			super(props);
+			this.state = {
+				loggedIn: false
+			};
+		}
+
+		componentDidMount() {
+			if (localStorage.getItem("user")) {
+				this.setState({ loggedIn: true });
+			} else {
+				this.setState({ loggedIn: false });
+			}
+		}
+
+		render() {
+			return this.state.loggedIn === true ? <Component /> : <Component2 />;
```

Comment:
```suggestion
			return this.state.loggedIn  ? <Component /> : <Component2 />;
```
Since this.state.loggedIn evaluates to true or false you don't need to check if it is === true